### ISSUE TITLE
v2.88.0 error for data.azurerm_kubernetes_cluster -> Error: setting `addon_profile`: addon_profile: '': source data must be an array or slice, got struct

### ISSUE THREAD

**Issue by mattduguid** [hashicorp/terraform-provider-azurerm#14455]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
terraform = 1.0.4
azurerm = 2.88.0 (broken) 2.87.0 (rolled back to and works)

### Affected Resource(s)
* `data.azurerm_kubernetes_cluster`

### Terraform Configuration Files (data source with issue)
```hcl
data "azurerm_kubernetes_cluster" "azure" {
  name = var.kubernetes_cluster_name
  resource_group_name = var.kubernetes_cluster_resource_group
}
```

### Terraform Configuration Files (addon_profile section of azurerm_kubernetes_cluster that data source has issue with)
```hcl
...
  addon_profile {
    azure_policy {
      enabled = false
    }
    http_application_routing { 
      enabled = false 
    }
    oms_agent {
      enabled = true
      log_analytics_workspace_id = "${data.azurerm_log_analytics_workspace.common.id}"
    }
  }
...
```

### Debug Output
2021-12-03T00:11:59.1025888Z 2021-12-03T00:11:59.099Z [ERROR] provider.terraform-provider-azurerm_v2.88.0_x5: setting state: addon_profile: '': source data must be an array or slice, got struct: timestamp=2021-12-03T00:11:59.099Z

### Expected Behaviour
data source to work

### Actual Behaviour
the above error in debug output

### Steps to Reproduce
1. `terraform apply`

**Comment by eehret** [comment]
Experiencing exactly the same issue here.

**Comment by aristosvo** [comment]
Hi @mattduguid! Thanks for raising this issue!

Could you give us more details on which `addon_profile` features are used on the `azurerm_kubernetes_cluster`? None, a few new ones, is the cluster maybe provisioned by hand?

**Comment by mattduguid** [comment]
@aristosvo updated in the initial notes

**Comment by stephybun** [comment]
Fixed by #14469 and released in v2.88.1.
